ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -158,36 +160,41 @@ startChainSyncConnectionThread ::
   SharedResources m ->
   ChainSyncResources m ->
   SchedulerConfig ->
+  StrictTVar m (Map PeerId (StrictTVar m TestFragH)) ->
   m (StrictTVar m (Maybe ChainSyncException))
-startChainSyncConnectionThread registry tracer cfg chainDbView fetchClientRegistry SharedResources {srPeerId, srCandidateFragment} ChainSyncResources {csrServer} SchedulerConfig {scChainSyncTimeouts} = do
+startChainSyncConnectionThread registry tracer cfg chainDbView fetchClientRegistry SharedResources {srPeerId} ChainSyncResources {csrServer} SchedulerConfig {scChainSyncTimeouts} varCandidates = do
   chainSyncException <- uncheckedNewTVarM Nothing
   _ <- forkLinkedThread registry ("ChainSyncClient" <> condense srPeerId) $
     bracketSyncWithFetchClient fetchClientRegistry srPeerId $ do
-      res <- try $ runConnectedPeersPipelinedWithLimits
-        createConnectedChannels
-        nullTracer
-        codecChainSyncId
-        chainSyncNoSizeLimits
-        (timeLimitsChainSync scChainSyncTimeouts)
-        (chainSyncClientPeerPipelined (basicChainSyncClient tracer cfg chainDbView srCandidateFragment))
-        (chainSyncServerPeer csrServer)
-      case res of
-        Left exn -> do
-          atomically $ writeTVar chainSyncException $ Just $ ChainSyncException srPeerId exn
-          case fromException exn of
-            Just (ExceededSizeLimit _) ->
-              traceUnitWith tracer ("ChainSyncClient " ++ condense srPeerId) "Terminating because of size limit exceeded."
-            Just (ExceededTimeLimit _) ->
-              traceUnitWith tracer ("ChainSyncClient " ++ condense srPeerId) "Terminating because of time limit exceeded."
-            Nothing ->
-              pure ()
-          case fromException exn of
-            Just ThreadKilled ->
-              traceUnitWith tracer ("ChainSyncClient " ++ condense srPeerId) "Terminated by GDD governor."
-            _ ->
-              pure ()
-        Right _ -> pure ()
+      bracketChainSyncClient nullTracer chainDbView varCandidates srPeerId ntnVersion $ \ varCandidate -> do
```

Comment:
I'm not too concerned with the change. Any hint of what `bracketChainSyncClient` does and why it wasn't needed so far?

Reply:
it adds and removes the peer to/from the candidates (and later handles) vars. we just inlined it so far.
it also starts some block validity monitor thing, I don't think it does much for us